packages/fuels-test-helpers/src/lib.rs

```diff
@@ -188,11 +188,9 @@ pub async fn setup_test_client(
     let config_with_coins: Value =
         serde_json::from_str(result.as_str()).expect("Failed to build config_with_coins JSON");
 
-    let srv_address = if let Some(node_config) = node_config {
-        node_config.addr
-    } else {
-        let free_port = pick_unused_port().expect("No ports free");
-        SocketAddr::new("127.0.0.1".parse().unwrap(), free_port)
+    let srv_address = match node_config {
+        Some(config) if config.addr.port() != 0 => config.addr,
```

Comment:
What would happen if the `config.addr.port` is not 0 but is already used ?

Reply:
I think the user should handle that. He will get a message ``Error: Address already in use``. @digorithm what do you think?

Reply:
@hal3e  and I analyzed this situation "What would happen if the config.addr.port is not 0 but is already used ?". Due to the way we run nodes using fuel-core.bin we can't catch `` Error: Address already in use`` coming from the server. The error would only be printed on the terminal but did not affect the test flow. Therefore, I added another condition that solved that bug.